expression.go

```diff
@@ -92,26 +109,26 @@ func And(c ...Expression) ExpressionGroup {
 
 /*String stringifies expressions for easy debugging*/
 func (c ExpressionGroup) String() string {
-	s, _, _ := c.construct(0, true)
+	s, _, _, _ := c.construct(0, true)
 	return s
 }
 
 /*********************************************************************************/
 /******************************** Negation Expression ****************************/
 /*********************************************************************************/
 
-func (n negation) construct(counter uint, topLevel bool) (string, map[string]interface{}, uint) {
-	s, m, c := n.expression.construct(counter, topLevel)
+func (n negation) construct(counter uint, topLevel bool) (string, map[string]*string, map[string]interface{}, uint) {
+	s, names, m, c := n.expression.construct(counter, topLevel)
```

Comment:
nit: in other similar places you kept this variable as `n`. not sure if it adds additional clarity here